Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM ruby:2.3.0
+RUN apt-get update -qq && apt-get install -y build-essential libpq-dev libgmp3-dev
+
+# gpg keys listed at https://github.com/nodejs/node
+RUN set -ex \
+  && for key in \
+    9554F04D7259F04124DE6B476D5A82AC7E37093B \
```

Comment:
Which part exactly?

So, thinking on a container way, this is the Dockerfile, with this file we can build **images** which in an OO analogy is a *class*. With this generated image (which is hosted on hub.docker) we can run **containers** which in the same analogy will be our *objects*.

To run containers, all dependencies needs to be installed into the image previously, which is being done by the Dockerfile, in our case ruby+node+things to compile gems/node modules.